queue/queue.go

```diff
@@ -1075,7 +1075,7 @@ func deliver(log mlog.Log, resolver dns.Resolver, m Msg) {
 			}
 			ourHostname = transport.Socks.Hostname
 		}
-		recipientDomainResult, hostResults = deliverDirect(qlog, resolver, dialer, ourHostname, transportName, msgs, backoff)
+		recipientDomainResult, hostResults = deliverDirect(qlog, resolver, dialer, ourHostname, transportName, transport, msgs, backoff)
```

Comment:
shouldn't we be passing transport.Direct here? it can just be nil. it will be clearer in the definition of deliverDirect() what can be expected of the transport (only a *config.TransportDirect).